experimental/navigation/hooks/useNativeNavigationTracker.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * will track navigation change events
+ * react-native-navigation
+ */
+
+import {ForwardedRef, useEffect, useMemo, useRef} from "react";
+
+import {TNativeNavigationContainer} from "../types/navigation";
+import spanCreator, {spanEnd} from "../otel/spanCreator";
+import {TracerRef} from "../otel/hooks/useTrace";
+import useSpan from "../otel/hooks/useSpan";
+
+type NativeNavRef = TNativeNavigationContainer;
+
+const useNativeNavigationTracker = (
+  ref: ForwardedRef<NativeNavRef>,
+  tracer: TracerRef,
+) => {
+  const navigationElRef = useMemo(() => {
+    const isMutableRef = ref !== null && typeof ref !== "function";
+    return isMutableRef ? ref.current : undefined;
+  }, [ref]);
+
+  const navView = useRef<string | null>(null);
+
+  // Initializing a Span
+  const span = useSpan();
+
+  useEffect(() => {
+    if (!navigationElRef) {
+      // do nothing in case for some reason there is no navigationElRef
+      return;
+    }
+
+    navigationElRef.registerComponentDidAppearListener(({componentName}) => {
+      if (!componentName) {
+        // do nothing in case for some reason there is no route
+        return;
+      }
+
+      spanCreator(tracer, span, navView, componentName);
+    });
+
+    navigationElRef.registerComponentDidDisappearListener(({componentName}) => {
+      if (!componentName) {
+        // do nothing in case for some reason there is no route
+        return;
+      }
+
+      spanEnd(span);
```

Comment:
I forget if we already discussed this but we should think about how we want to handle the last view when the app ends or gets backgrounded. If we don't have a strong opinion maybe just adding a function to the hook's return value like `endLastSpan` or something and telling the user that if they care about ending it they should hook it up to a handler

Reply:
we talked about it, but we didn't decide any approach. I like the suggestion, I'll dig into this on friday